process/main.py

```diff
@@ -521,21 +521,44 @@ def validate_input(self):
                     variables = line.strip().split(sep, 1)[0]
                     variables_in_in_dat.append(variables)
 
-        obs_vars = []
-        replace_hints = []
+        obs_vars_in_in_dat = []
+        replace_hints = {}
         for var in variables_in_in_dat:
-            new_var = obsolete_variables.get(var)
-            if new_var:
-                obs_vars.append(var)
-                replace_hints.append(new_var)
-
-        if len(replace_hints) > 0:
-            print(
-                "The IN.DAT file contains obsolete variables from the OBS_VARS dictionary."
-            )
-            print(
-                f"The obsolete variables in your IN.DAT file are: {obs_vars}. Please change them to the following corresponding new variable(s) before continuing: {replace_hints}."
+            if var in obsolete_variables:
+                obs_vars_in_in_dat.append(var)
+                new_var = obsolete_variables.get(var)
+                replace_hints[var] = new_var
```

Comment:
These two lines are a bit confusing and I think thats probably because of the the variable names. I think you could just do the following:

```suggestion
                replace_hints[var] = obsolete_variables.get(var)
```